flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -272,6 +272,49 @@ object ScalarOperatorGens {
             }
         }
 
+      // minus arithmetic of time points (i.e. for TIMESTAMPDIFF for TIMESTAMP_LTZ)
+      case (TIMESTAMP_WITH_LOCAL_TIME_ZONE, TIMESTAMP_WITH_LOCAL_TIME_ZONE |
+                TIMESTAMP_WITHOUT_TIME_ZONE | TIME_WITHOUT_TIME_ZONE | DATE) |
+           (TIMESTAMP_WITH_LOCAL_TIME_ZONE | TIMESTAMP_WITHOUT_TIME_ZONE |
+            TIME_WITHOUT_TIME_ZONE | DATE, TIMESTAMP_WITH_LOCAL_TIME_ZONE) if !plus =>
+        resultType.getTypeRoot match {
+          case INTERVAL_YEAR_MONTH =>
+            generateOperatorIfNotNull(ctx, resultType, left, right) {
+              (ll, rr) => (left.resultType.getTypeRoot, right.resultType.getTypeRoot) match {
+                case (TIMESTAMP_WITH_LOCAL_TIME_ZONE, TIMESTAMP_WITH_LOCAL_TIME_ZONE) =>
+                  val leftTerm = s"$ll.getMillisecond()"
+                  val rightTerm = s"$rr.getMillisecond()"
+                  s"${qualifyMethod(BuiltInMethods.SUBTRACT_MONTHS)}($leftTerm, $rightTerm)"
+                case _ =>
+                  throw new CodeGenException(String.format(
+                    "Unsupported TIMESTAMPDIFF(%s, %s, %s) which contains TIMESTAMP_LTZ type." +
+                      " TIMESTAMP_LTZ type only supports TIMESTAMPDIFF(timepointunit," +
```

Comment:
Simplify the error message a little bit: `TIMESTAMP_LTZ only supports diff between the same type.`